evm/v0.5/src/helpers.ts

```diff
@@ -603,24 +665,183 @@ export function generateSAID(sa: ServiceAgreement): Hash {
       `extracted wrong params: ${saParam} from coordinatorFactory.interface.functions.getId`,
     )
   }
-  return ethers.utils.solidityKeccak256(
+  return utils.solidityKeccak256(
     SERVICE_AGREEMENT_TYPES,
     serviceAgreementValues(sa),
   )
 }
 
+/**
+ * Generates the bytes to call and execute the oracleRequest() function
+ * @param sAID The service agreement ID
+ * @param callbackAddr The callback contract address
+ * @param callbackFunctionId The callback function id
+ * @param nonce The nonce
+ * @param data optional data
+ */
+export function executeServiceAgreementBytes(
+  sAID: string,
+  callbackAddr: string,
+  callbackFunctionId: any,
+  nonce: utils.BigNumberish,
+  data?: any,
+): any {
+  const types = [
+    'address',
+    'uint256',
+    'bytes32',
+    'address',
+    'bytes4',
+    'uint256',
+    'uint256',
+    'bytes',
+  ]
+  const values = [
+    ethers.constants.AddressZero,
+    0,
+    sAID,
+    callbackAddr,
+    callbackFunctionId,
+    nonce,
+    1,
+    data || '0x',
+  ]
+  const encodedArgs = utils.defaultAbiCoder.encode(types, values)
+  const funcSelector = encodeFunctionSignature(
+    'oracleRequest(address,uint256,bytes32,address,bytes4,uint256,uint256,bytes)',
+  )
+  return utils.concat([funcSelector, encodedArgs])
```

Comment:
Couldn't we get rid of this function altogether for `myContract.functions.oracleRequest.encode`? It should do what we need